cypress/integration/9-room-status-bar/account-suspension.ts

```diff
@@ -0,0 +1,113 @@
+            roomId = data;
+            cy.visit("/#/room/" + roomId);
+        });
+    });
+
+    afterEach(() => {
+        // Attempting to stop Synapse breaks the tests.
+        // Not sure what's going on here.
+        // cy.stopSynapse(synapse);
+    });
+
+    it("shouldn't display an error message when there is no error", () => {
+        // User sends message
+        cy.get(".mx_RoomView_body .mx_BasicMessageComposer_input").type(`${text}{enter}`);
+
+        // Wait for message to send
+        cy.get(".mx_RoomView_body .mx_EventTile").contains(".mx_EventTile[data-scroll-tokens]", text);
+
+        // Give an error a little time to show up. It shouldn't.
+        cy.wait(1_000);
+        cy.get(".mx_RoomStatusBar_unsentMessages .mx_RoomStatusBar_unsentTitle").should('not.exist');
+    });
+
+    it("should display a generic error if the error is not a user account suspension", () => {
+        sendWithErrorResponse(text, { errcode: "SOME_OTHER_ERROR" });
+        cy.get(".mx_RoomStatusBar_unsentMessages .mx_RoomStatusBar_unsentTitle")
+            .should('include.text', "Some of your messages have not been sent");
+        cy.get(".mx_RoomStatusBar_unsentMessages .mx_RoomStatusBar_unsentTitle")
+            .should('not.include.text', "Your account is suspended");
+    });
+
+    it("should display a generic user account suspended if no href is provided", () => {
+        sendWithErrorResponse(text, { errcode: userAccountSuspended });
+        cy.get(".mx_RoomStatusBar_unsentMessages .mx_RoomStatusBar_unsentTitle")
+            .should('include.text', "Your account is suspended");
+        cy.get(".mx_RoomStatusBar_unsentMessages .mx_RoomStatusBar_unsentTitle")
+            .should('include.text', "Please contact the administrator");
+        cy.get(".mx_RoomStatusBar_unsentMessages .mx_RoomStatusBar_unsentTitle")
+            .should('not.include.text', "To learn more, visit");
+    });
+
+    it("should display a user account suspended with a link if a href is provided", () => {
+        sendWithErrorResponse(text, {
+            errcode: userAccountSuspended,
+            href: href,
+        });
+        cy.get(".mx_RoomStatusBar_unsentMessages .mx_RoomStatusBar_unsentTitle")
+            .should('include.text', "Your account is suspended");
+        cy.get(".mx_RoomStatusBar_unsentMessages .mx_RoomStatusBar_unsentTitle")
+            .should('not.include.text', "Please contact the administrator");
+        cy.get(".mx_RoomStatusBar_unsentMessages .mx_RoomStatusBar_unsentTitle")
+            .should('include.text', "To learn more, visit");
+        cy.get(".mx_RoomStatusBar_unsentMessages .mx_RoomStatusBar_unsentTitle a")
+            .should('have.attr', "href").and('include', href);
+    });
+});
+
+// Pretend to send a message but inject an error response.
+function sendWithErrorResponse(text, response) {
+    cy.intercept("http://localhost:*/_matrix/client/r0/rooms/*/send/m.room.message/*", req => {
```

Comment:
sorry, I really think this needs to be an admin API call and not something we intercept. We should be testing that we behave correctly when Synapse tells us the account is suspended, not when we lie to it (otherwise we'd write an entirely different, less helpful, test).

Reply:
That feels like it would block an ongoing project.

Could we proceed as follows?

1. Land it with the intercept.
2. Perform the tests that we cannot perform until this feature has landed.
3. Design an admin API.
4. Spend a few weeks (or however long is required) bikeshedding, implementing and landing the admin API(s) with back-end.
5. Once we have an admin API, add a second test (or replace the existing test with a new one) that uses the admin API?

Reply:
It would block the project, yes. Landing partial or incomplete code isn't viable as an option: we've literally never returned to temporary code, despite several promises that we will. We no longer take the chance, sorry.